ingestion/src/metadata/ingestion/source/pipeline/openlineage/metadata.py

```diff
@@ -0,0 +1,534 @@
+            try:
+                schema_fqn = self._get_schema_fqn_from_om(table_details.schema)
+
+                return f"{schema_fqn}.{table_details.name}"
+            except FQNNotFoundException:
+                return None
+
+    def _get_entity_fqn_from_om(
+        self, entity_type: Union[type(Table), type(DatabaseSchema)], **kwargs
+    ) -> Optional[str]:
+        fully_qualified_name = fqn.build(
+            metadata=self.metadata, entity_type=entity_type, **kwargs
+        )
+        if fully_qualified_name:
+            entity: Union[Table, DatabaseSchema] = self.metadata.get_by_name(
+                entity=entity_type, fqn=fully_qualified_name
+            )
+            if entity:
+                result = entity.fullyQualifiedName.__root__
+                return result
+
+        return None
+
+    def _get_table_fqn_from_om(self, table_details: TableDetails) -> Optional[str]:
+        """
+        Based on partial schema and table names look for matching table object in open metadata.
+        :param schema: schema name
+        :param table: table name
+        :return: fully qualified name of a Table in Open Metadata
+        """
+        result = None
+        services = self.source_config.dbServiceNames
+        for db_service in services:
+            result = self._get_entity_fqn_from_om(
+                Table,
+                service_name=db_service,
+                table_name=table_details.name,
+                schema_name=table_details.schema,
+                database_name=None,
+            )
+            if result:
+                return result
+        if not result:
+            raise FQNNotFoundException(
+                f"Table FQN not found for table: {table_details} within services: {services}"
+            )
+        return result
+
+    def _get_schema_fqn_from_om(self, schema: str) -> Optional[str]:
+        """
+        Based on partial schema name look for any matching DatabaseSchema object in open metadata.
+
+        :param schema: schema name
+        :return: fully qualified name of a DatabaseSchema in Open Metadata
+        """
+        result = None
+        services = self.source_config.dbServiceNames
+
+        for db_service in services:
+            databases = [
+                x.get("name")
+                for x in self.metadata.client.get(
+                    "/databases",
+                    data={
+                        "service": db_service,
+                        "include": "non-deleted",
+                        "limit": 1000000,
+                    },
+                ).get("data", [])
+            ]
+
+            for database in databases:
+                result = self._get_entity_fqn_from_om(
+                    DatabaseSchema,
+                    service_name=db_service,
+                    schema_name=schema,
+                    database_name=database,
+                )
+
+                if result:
+                    return result
```

Comment:
This logic may lead to a lot of request in case when the number of databases are too high I would recommend making call to ES instead. The similar way how we are doing it for tables which can reduce the number of request drastically from large number of requests to a single ES call

Reply:
I agree but I think the right place to fix it would be extending fqn.build for DatabaseSchema with support to querying ES (same as can be done with Table entities) - which should be possible after introducing separate index for DatabaseSchema

Reply:
extended fqn module to support `skip_es_search` for DatabaseSchema